Source/Curves/ContinuousCurves/SMCurveClass.f90

```diff
@@ -565,8 +597,148 @@ double precision function fmin(self,ax,bx,tol)
 !
 !  end of main loop
 !
-  190 fmin=x
+  190 fminFMM=x
       return
-      END FUNCTION fmin
+      END FUNCTION fminFMM
+!
+!//////////////////////////////////////////////////////////////////////// 
+! 
+   FUNCTION fMin(self, aIn, bIn, tol, pnt, nHat ) RESULT(z)
+      IMPLICIT NONE
+!
+!     ---------
+!     Arguments
+!     ---------
+!
+      CLASS(SMCurve)          :: self
+      REAL(KIND=RP)           :: aIn, bIN, tol
+      REAL(KIND=RP)           :: z
+      REAL(KIND=RP), OPTIONAL :: pnt(3), nHat(3)
+!
+!     ---------------
+!     local Variables
+!     ---------------
+!
+      REAL(KIND=RP) :: a, b, c, d, e, m, p, q, r, t2, u, v, w, fu, fv, fw, fx, t, x
+      REAL(KIND=RP) :: t3, eps
+!
+!     -----
+!     Setup
+!     -----
+!
+      eps = EPSILON(1.0_RP)
+      t3  = tol/3.0_RP
+      eps = SQRT(eps)
+      
+      c  = 0.5_RP*(3.0_RP - SQRT(5.0_RP))
+      a  = aIn
+      b  = bIn
+      x  = a + c*(b-a)   ; v = x  ; w = x   ; e = 0.0_RP; d = 0.0_RP
+      
+      IF ( PRESENT(pnt) )     THEN
+         fx = distanceSquared(x,self,pnt,nHat) 
+      ELSE 
+         fx = ObjectiveFunction(self,x)
+      END IF 
+      fv = fx; fw = fx
+!
+!     ---------
+!     Main loop
+!     ---------
+!
+      DO
+         m  = 0.5_RP*(a + b)
+         t  = eps*ABS(x) + t3
+         t2 = 2.0_RP*t
+         
+         IF ( ABS(x-m) <= t2 - 0.5_RP*(b-a) )     EXIT
+         
+         p = 0.0_RP; q = 0.0_RP; r = 0.0_RP
+         
+         IF ( ABS(e) > t )     THEN !Fit parabola
+            r = (x - w)*(fx - fv)
+            q = (x - v)*(fx - fw)
+            p = (x - v)*q - (x - w)*r
+            q = 2.0_RP*(q - r)
+            IF ( q > 0 )     THEN
```

Comment:
The original logic in the FMM implementation is
```fortran
   50 if ((dabs(p).ge.dabs(0.5d0*q*r)).or.(p.le.q*(a-x))&
     &          .or.(p.ge.q*(b-x))) go to 60
```
So applying De Morgan this should flip around all the logic and turn the `OR`s into `AND`s
```fortran
         IF ( ABS(p) < ABS(0.5_RP*q*r) .AND. &
              p > q*(a - x)            .AND. &
              p < q*(b - x) )                    THEN !Parabolic interpolation step
```

Reply:
I didn't try sorting the goto logic of FMM, but was following Brent. I may have gotten it mixed up, however. I'll look again. I'll send it by email since I can't seem to attach it here.

Reply:
Interesting. You correctly implemented the version from Brent. I wonder why flipping the FMM logic gives something different.

Reply:
On all the meshes I have tested either logic gives the same results. So I would say we go with the original algorithm as given by Brent.

Reply:
I agree.